openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
+            for (var w = 0; w < themeKeys.length; w++) {
+                var tempThemeName = themeKeys[w];
+                tempElementValue = tempElementValue + themeObj[v][tempThemeName];
+            }
+            subIndex[themeObjMunic] = tempElementValue;
         }
-    });
-
-    map.addControl(layerControl.setPosition('topleft'));
-    // TODO set the map max zoom to 9
-    // The interactivity of the map/charts will not work with a map zoom greater then 9
-
-    // Add layers form tilestream list
-    $(document).ready(function() {
-        $('#addLayer').click(function() {
-            var e = document.getElementById('layer-list');
-            var layerId = e.options[e.selectedIndex].value;
-
-            // Look up the project definition layer id using the layer name
-            var selectedPDefArray = projectDefinition[layerId];
-            var selectedPDefStr = selectedPDefArray.toString();
-
-            // TODO remove this link and replace with Django api call
-            // Link to Github is a temp proof of concept
-            // Load the project definition json
-            selectedPDefStr = 'https://api.github.com/repos/bwyss/oq-platform/git/blobs/9864cb7a5d36572af1451e585f4516d1d19ce568?callback=_processGithubResponse';
-
-            $.getJSON(selectedPDefStr+'?format=json&callback=?', function(pdJson) {
-                encodedData = pdJson.data.content;
-                selectedPDef = window.atob(encodedData);
-                pdData = JSON.parse(selectedPDef);
-                loadPD(selectedPDef);
-            });
-
-            // Look up the tile layer id using the layer name
-            var layerIdArray = layerNames[layerId];
-            var selectedLayer = layerIdArray.toString();
-            var hasGrid = $.inArray(selectedLayer, layerGrids) > -1;
-
-            // Check for duplicae layes
-            if (selectedLayer in layers) {
-                showDuplicateMsg();
-            }
-            else {
-                map.removeLayer(utfGrid);
-                utfGrid = {};
-
-                var tileLayer = L.tileLayer(TILESTREAM_URL+
-                    selectedLayer+
-                    '/{z}/{x}/{y}.png',{wax: TILESTREAM_URL+
-                    selectedLayer+
-                    '.json'});
-                layerControl.addOverlay(tileLayer, selectedLayer);
-                map.addLayer(tileLayer);
-                // Keep track of layers that have been added
-                layers[selectedLayer] = tileLayer;
-
-                if (hasGrid == true) {
-                    gridList = 1;
-                    utfGrid = new L.UtfGrid(TILESTREAM_URL+
-                        selectedLayer+
-                        '/{z}/{x}/{y}.grid.json?callback={cb}', {Default: false, JsonP: false});
-                    map.addLayer(utfGrid);
-                    utfGridClickEvent(utfGrid);
+    } else if (operator == 'Weighted sum') {
+        for (var v1 = 0; v1 < themeObj.length; v1++) {
```

Comment:
I guess a previous comment was hidden by github after some changes in the code around it. I don't see the point of using a different name for the variables in completely separated loops (v1, v2, vN). Couldn't you just use v instead?